google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -89,7 +89,8 @@ class MessageDispatcher {
   private final LinkedBlockingQueue<AckRequestData> pendingAcks = new LinkedBlockingQueue<>();
   private final LinkedBlockingQueue<AckRequestData> pendingNacks = new LinkedBlockingQueue<>();
   private final LinkedBlockingQueue<AckRequestData> pendingReceipts = new LinkedBlockingQueue<>();
-
+  private final ConcurrentHashMap<String, ReceiptCompleteData> outstandingReceipts =
```

Comment:
We aren't making use of the concurrent features of this map, so let's just make it a regular map.

Reply:
Have changed it to a linkedhashmap to maintain insertion order